app/javascript/components/common/Pagination.tsx

```diff
@@ -19,6 +19,10 @@ export function Pagination({
     return null
   }
 
+  current = Number(current)
+  around = Number(around)
+  total = Number(total)
```

Comment:
The bit below this is fine. It's just this bit that is "controversial".

Reply:
What's likely happening is that the URL Param page isn't actually converted to a number when it's read, and that's what you see as "string value".

The better solution would be, if that's the problem, to type convert that value to a number instead of here.

That said, using Number to ensure its a number is the right thing and you definitely.dont want the parseXXX family because you know that it's a number and it should break when it's not. parseXXX doesn't break and gives you unexpected values instead.

Reply:
> The better solution would be, if that's the problem, to type convert that value to a number instead of here.

Yeah - agreed. What's confusing me is that it is definitely a number in the JSON coming out of Rails and we pretty much just read that here as I understand it. @kntsoriano Could you try and trace it through to see where this is coming from pls. I don't mind leaving this here as a final guard (maybe just "current" and "total") - but it would be good to understand the underlying problem pls.

Reply:
Again I don't think the issue is the JSON, but the URL param override.

Can you check that one?

Reply:
I don't think we read from the URL param. As far as I can tell in the code, we read from the JSON when the page loads.

This seems to be the code that does that.

<img width="556" alt="Screenshot 2021-06-21 at 14 54 27" src="https://user-images.githubusercontent.com/286476/122773865-9cab3b80-d2a0-11eb-8cbb-34e926a12e06.png">

I don't know how any of this works though, and I may well be missing a pathway that does what you're saying, so @kntsoriano will need to check 🙂

Reply:
(These were all just educated guesses without looking at the code. Might as well be something different, but it's were I would look first)